Source/gg2/Objects/Menus/Hosting Menu Elements/AdvHostOptionsController.events/Create.xml

```diff
@@ -13,17 +13,43 @@
       <not>false</not>
       <appliesTo>.self</appliesTo>
       <arguments>
-        <argument kind="STRING">menu_create(48, 172, 500, 260, 32, 40, 112, 128);
+        <argument kind="STRING">ini_open("gg2.ini");
+global.mapRotationFile = ini_read_string("Server", "MapRotation", "");
+ini_close();
+
+menu_create(48, 172, 500, 260, 32, 40, 112, 128);
 menumode = true;
 
 menu_background(512, 24, 8, 12, 4);
 bgtabs = true;
 
+var apos; // dumb workaround
+apos = "'";
+
 menu_addedit_num("Hosting Port:", "global.hostingPort", '
     gg2_write_ini("Settings", "HostingPort", argument0);
 ', 65535);
-var apos; // dumb workaround
-apos = "'";
+
+if (global.mapRotationFile == "") showMapRot = "Default";
+else showMapRot = string_replace(filename_name(global.mapRotationFile), ".txt", "");
+menu_addlink_label("Map Rotation:", "showMapRot", '
+
+    var absPath;
+
+    absPath = get_open_filename("Map Rotation|*.txt", "");
+
+    if (absPath == "") {
+        showMapRot = "Default";
+        gg2_write_ini("Server", "MapRotation", "");
+        exit;
+    }
+    
+    showMapRot = filename_name(absPath);
+        
+    gg2_write_ini("Server", "MapRotation", getRelativePath(program_directory, absPath));
```

Comment:
I'm still not 100% sure, but it looks like I was right, [`ini_open`](https://docs.yoyogames.com/source/dadiospice/002_reference/file%20handling/ini%20files/ini_open.html) uses relative paths, because usually the file is right next to the exe, while `get_open_flename` returns an absolute path